nexus/external-api/src/lib.rs

```diff
@@ -3446,6 +3453,34 @@ pub trait NexusExternalApi {
         path_params: Path<params::ProbePath>,
     ) -> Result<HttpResponseDeleted, HttpError>;
 
+    // Audit logging
+
+    /// View audit log
+    ///
+    /// A single item in the audit log represents both the beginning and
+    /// end of the logged operation (represented by `time_started` and
+    /// `time_completed`) so that clients do not have to find multiple entries
+    /// and match them up by request ID to get the full picture of an operation.
+    /// Because timestamps may not be unique, entries have also have a unique
+    /// `id` that can be used to deduplicate items fetched from overlapping
+    /// time intervals.
+    ///
+    /// Note that entries are immutable and are ordered by `time_completed`,
+    /// not `time_started`, in order to ensure stable results. If you request
+    /// entries from `t0 <= time_completed < t1` and `t1` is in the past, you
+    /// can be confident that the resulting list is complete, i.e., requesting
+    /// the same range again later will always produce the same list. We do not
+    /// include items in the audit log response until they are marked completed.
```

Comment:
I read this comment before getting to the one in `nexus/db-queries/src/db/datastore/audit_log.rs` and was a bit lost. Once I read the comment in the other file, I understood; maybe throw a pointer to that comment here for further explanation? The example in that comment is helpful and made things clear for me.

Reply:
I kinda went the other way, saying a bit less to raise fewer questions, focusing on the guarantees from a practical POV and less on the underlying reason for it.